hibernate-core/src/test/java/org/hibernate/test/bytecode/enhancement/lazy/proxy/inlinedirtychecking/DirtyCheckPrivateUnMappedCollectionTest.java

```diff
@@ -45,11 +48,21 @@ protected void configureStandardServiceRegistryBuilder(StandardServiceRegistryBu
 
 	@Override
 	protected void applyMetadataSources(MetadataSources sources) {
-		sources.addAnnotatedClass( Measurement.class );
+		String byteCodeProvider = Environment.getProperties().getProperty( AvailableSettings.BYTECODE_PROVIDER );
+		if ( byteCodeProvider != null && !Environment.BYTECODE_PROVIDER_NAME_BYTEBUDDY.equals( byteCodeProvider ) ) {
```

Comment:
testing against `javaassist` might be more accurate and maintainable. The above logic is based on the precondition that the only non-bytebuddy provider is `javaassist`, which might not be correct all the time.